perun/utils/common/common_kit.py

```diff
@@ -437,4 +451,97 @@ def get_module(module_name: str) -> types.ModuleType:
     return MODULE_CACHE[module_name]
 
 
+def compact_convert_list_to_str(
+    number_list: list[int | float] | array.array[float] | array.array[int], float_precision: int = 2
+) -> list[str]:
+    """Converts list to list of compact strings
+
+    :param number_list: list of numbers
+    :param float_precision: float precision of the numbers
+    :return list of compact strings
+    """
+    return [compact_convert_num_to_str(num, 2) for num in number_list]
+
+
+def compact_convert_num_to_str(number: int | float, float_precision: int = 2) -> str:
+    """Converts the number to the smallest string possible
+
+    :param number: converted number
+    :param float_precision: float precision, i.e. number of decimal places in number
+    :return compact string
+    """
+    return str(to_compact_num(number, float_precision))
+
+
+def to_compact_num(number: int | float, float_precision: int = 2) -> int | float:
+    """Converts the number to the smallest string possible
+
+    :param number: converted number
+    :param float_precision: float precision, i.e. number of decimal places in number
+    :return compact num
+    """
+    if isinstance(number, int):
+        return number
+    elif number.is_integer():
+        return int(number)
+    else:
+        return round(number, float_precision)
+
+
+@contextlib.contextmanager
+def disposable_resources(disposable: Any) -> Any:
+    """Helper context manager that disposes of the passed object
+
+    :param disposable: object that should be disposed immediately
+    :return: object
+    """
+    try:
+        yield disposable
+    except Exception:
+        # Re-raise the encountered exception
+        raise
+    finally:
+        del disposable
+        gc.collect()
+
+
+def binary_search(
+    values: list[Any], value: Any, low: int, high: int, key: Callable[[Any], Any] = lambda x: x
+) -> int:
+    """Classical binary search algorithm
```

Comment:
Could maybe https://docs.python.org/3/library/bisect.html be used here? I'm not sure whether the interface is compatible with what you need.

Reply:
This is neat, I actually didn't assume there would be such algorithm in standard library, and it should be usable here :)